### PR TITLE
NOD | Add confirmation update unit tests

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > Adding unit tests and minor fixes to new NOD confirmation page (behind feature flag)
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Decision Reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#81507](https://github.com/department-of-veterans-affairs/va.gov-team/issues/81507)

## Testing done

- _Describe what the old behavior was prior to the change_
  > No unit test for new components/functions
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

N/A

## What areas of the site does it impact?

Notice of Disagreement

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by anniebtran — COMMENTED** [review]
still got a few things left to review but wanted to see if you had thoughts on any of my comments so far

**Review by anniebtran — COMMENTED** [review]
Okay everything else looking pretty good! Lmk what your thoughts are on my other comments, but otherwise I can approve

File: src/applications/appeals/10182/tests/containers/ConfirmationPage.unit.spec.jsx [hunk 1]
```diff
@@ -11,17 +11,27 @@ import formConfig from '../../config/form';
...
   user: {
     profile: {
       userFullName: renderName
         ? { first: 'Foo', middle: 'Man', last: 'Choo', suffix }
         : {},
     },
   },
+  featureToggles: {
+    loading: false,
+    nodConfirmationUpdate,
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/10182/tests/containers/ConfirmationPage.unit.spec.jsx:31]: I don't remember why, but the feature toggle API response includes both snake & camel case toggles. In this situation, we only need the snake-cased version. I could remove the camel-case variable if it's confusing.

File: src/applications/appeals/shared/components/ConfirmationPersonalInfo.jsx [hunk 2]
```diff
@@ -6,25 +6,21 @@ import { getPhoneString } from '~/platform/forms-system/src/js/utilities/data/pr
 import { renderFullName, maskVafn } from '../utils/data';
 import { getReadableDate } from '../utils/dates';
 
-const ConfirmationPersonalInfo = ({ profile, data }) => {
-  const { userFullName, dob } = profile;
-  const { veteran } = data;
+const ConfirmationPersonalInfo = ({ profile, data } = {}) => {
+  const { userFullName, dob } = profile || {};
+  const { veteran = {} } = data || {};
+  const { address = {} } = veteran;
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/shared/components/ConfirmationPersonalInfo.jsx]: Thanks! Fixed.

File: src/applications/appeals/10182/tests/containers/ConfirmationPage.unit.spec.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (anniebtran) [reply src/applications/appeals/10182/tests/containers/ConfirmationPage.unit.spec.jsx:31]: Ah okay interesting 🤔  If that's what the data looks like in the API response, it's probably fine to leave it as is. Thanks for clarifying!!

**Review by anniebtran — APPROVED** [review]
LGTM!

**Pull request merged**